src/mem.rs

```diff
@@ -0,0 +1,60 @@
+// NOTE Copied verbatim from the rlibc crate
```

Comment:
I think I missed your point before. It makes sense to put rlibc below this crate to have to fix bugs in a single place. To do that rlibc would have to gain a "weak" Cargo feature that marks the `mem*` functions as weak symbols.

> there should be some way to use rlibc without forcing a crates.io download

Probably just make it a submodule of rust-lang/rust. Like this crate/repository will.